### PR TITLE
Subscription and Topic Management and Service Version

### LINKED ISSUE #12053: Add Topics/Subscriptions support to Management API
Add support for topic/subscription update and creation.

### PR DESCRIPTION
**Pull request by conniey**
* Adds support for Subscription and Topic management operations
* Fixes visibility of QueueDescription parameters and adds EntityHelper to set internal methods similar to approach used in:
  * https://github.com/teamfx/openjfx-9-dev-rt/blob/master/modules/javafx.graphics/src/main/java/com/sun/javafx/scene/ParentHelper.java
* Updates models for Subscription and Topic
* Adds ServiceBusServiceVersion

Fixes #12053

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementAsyncClient.java [hunk 1]
```diff
@@ -342,48 +792,363 @@
     }
 
     /**
-     * Creates a queue with its context.
+     * Updates a subscription with the given {@link SubscriptionDescription}. The {@link SubscriptionDescription} must
+     * be fully populated as all of the properties are replaced. If a property is not set the service default value is
+     * used.
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementAsyncClient.java:797]: Is there a online documentation which we can point user to for list of properties and default values ? More methods below have same comment.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementAsyncClient.java [hunk 2]
```diff
@@ -537,14 +1407,92 @@
...
+    Mono<Response<SubscriptionDescription>> updateSubscriptionWithResponse(SubscriptionDescription subscription,
+        Context context) {
+        if (subscription == null) {
+            return monoError(logger, new NullPointerException("'subscription' cannot be null"));
+        }
+
+        final String topicName = subscription.getTopicName();
+        final String subscriptionName = subscription.getSubscriptionName();
+        if (topicName == null || topicName.isEmpty()) {
+            return monoError(logger, new IllegalArgumentException("'getTopicName' cannot be null or empty."));
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementAsyncClient.java:1436]: `topicName == null` : NullPointerException ? and add java doc for it.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementAsyncClient.java [hunk 3]
```diff
@@ -537,14 +1407,92 @@
...
+        } else if (subscriptionName == null || subscriptionName.isEmpty()) {
+            return monoError(logger, new IllegalArgumentException("'getSubscriptionName' cannot be null or empty."));
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementAsyncClient.java:1438]: `subscriptionName == null ` : NullPointerException ?

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementAsyncClient.java [hunk 4]
```diff
@@ -537,14 +1407,92 @@
...
+     *     on the updated entity. Any values not provided are set to the service default values.
+     * @param context Context to pass into request.
+     *
+     * @return A Mono that completes with the updated {@link TopicDescription}.
+     */
+    Mono<Response<TopicDescription>> updateTopicWithResponse(TopicDescription topic, Context context) {
+        if (topic == null) {
+            return monoError(logger, new NullPointerException("'topic' cannot be null"));
+        } else if (topic.getName() == null || topic.getName().isEmpty()) {
+            return monoError(logger, new IllegalArgumentException("'topic.getName' cannot be null or empty."));
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementAsyncClient.java:1474]: `topic.getName() == null ` : NullPointerException ?

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/SubscriptionDescription.java [hunk 5]
```diff
@@ -175,6 +181,117 @@
...
+     * Json deserialization constructor.
+     */
+    @JsonCreator
+    SubscriptionDescription() {
+    }
+
+    /**
+     * Creates an instance with the name of the subscription and its associated topic. Default values for the
+     * subscription are populated. The properties populated with defaults are:
+     *
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/SubscriptionDescription.java:212]: Should we put default values as well in documentation , so user know what are the values?.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/TopicDescription.java [hunk 6]
```diff
@@ -219,6 +223,81 @@ private AuthorizationRulesWrapper(
...
+    /**
+     * Json deserialization constructor.
+     */
+    @JsonCreator
+    TopicDescription() {
+    }
+
+    /**
+     * Creates an instance with the name of the topic. Default values for the topic are populated. The properties
+     * populated with defaults are:
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/TopicDescription.java:242]: same here, should we show to user what are default values ?

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/TopicDescription.java [hunk 7]
```diff
@@ -635,7 +714,7 @@ public Boolean isEnableSubscriptionPartitioning() {
      * @param enableSubscriptionPartitioning the enableSubscriptionPartitioning value to set.
      * @return the TopicDescription object itself.
      */
-    public TopicDescription setEnableSubscriptionPartitioning(Boolean enableSubscriptionPartitioning) {
+    TopicDescription setEnableSubscriptionPartitioning(Boolean enableSubscriptionPartitioning) {
         this.enableSubscriptionPartitioning = enableSubscriptionPartitioning;
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/TopicDescription.java:718]: should this API be called `enableSubscriptionPartitioning`

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/TopicDescription.java [hunk 8]
```diff
@@ -657,7 +736,7 @@ public Boolean isEnableExpress() {
      * @param enableExpress the enableExpress value to set.
      * @return the TopicDescription object itself.
      */
-    public TopicDescription setEnableExpress(Boolean enableExpress) {
+    TopicDescription setEnableExpress(Boolean enableExpress) {
         this.enableExpress = enableExpress;
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/TopicDescription.java:740]: setEnableExpress -> enableExpress()

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementAsyncClient.java (hunk 1 above)
Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementAsyncClient.java:797]: The likelihood that users click on the documentation first before trying this out and having it fail is low.
There is a `@see` annotation to each of these methods if they want more complete information.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementAsyncClient.java (hunk 2 above)
Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementAsyncClient.java:1436]: There is a null check when creating the actual subscription object. So this won't happen. I'll remove the double checks.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/SubscriptionDescription.java (hunk 5 above)
Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/SubscriptionDescription.java:212]: I'll add those.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/TopicDescription.java (hunk 8 above)
Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/TopicDescription.java:740]: This is package-private, so it doesn't matter.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/TopicDescription.java (hunk 7 above)
Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/TopicDescription.java:718]: This is package-private, so it doesn't matter.

**Pull request merged**